assignments/prototypes.js

```diff
@@ -102,9 +144,84 @@
   console.log(archer.greet()); // Lilith offers a greeting in Elvish.
   console.log(mage.takeDamage()); // Bruce took damage.
   console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
+//*/
 
   // Stretch task: 
   // * Create Villain and Hero constructor functions that inherit from the Humanoid constructor function.  
-  // * Give the Hero and Villains different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
-  // * Create two new objects, one a villain and one a hero and fight it out with methods!
\ No newline at end of file
+  // * Give the Hero and Villains different methods that could be used to remove
+   //  health points from objects which could result in destruction if health gets to 0 or drops below 0;
+  // * Create two new objects, one a villain and one a hero and fight it out with methods!
+
+  function determineDamager(target) {
+    let damage = Math.floor(Math.random() * Math.ceil(target.healthPoints+1));
+    return ( Math.ceil(damage) );
+  }
+
+  function Villain(attrs) {
+    Humanoid.call(this, attrs);
+    this.isVillan = attrs.isVillan;
+  }
+  Villain.prototype = Object.create(Humanoid.prototype);
+  Villain.prototype.assault = function attack(obj) {
+    let damage = determineDamager(obj);
+    obj.healthPoints -= damage;
+    if(obj.healthPoints <= 0) { return ( [ `${this.name} murders ${obj.name}!`, obj.destroy() ] ); }
+    if ( damage <= 0 ) { return ( `${this.name} stubs his toe and curses!` ); }
+    if (damage === 1) { return ( `${obj.name} is assailed for ${damage} point of damage.` ); }
+    return ( `${obj.name} is assailed for ${damage} points of damage.` );
+  }
```

Comment:
this is a great bit of logic for the attack.
remember you can omit a lot of this code with one-line statements.

```
    if(obj.healthPoints <= 0) return `${this.name} murders ${obj.name}!`, obj.destroy();  
    if ( damage <= 0 ) return `${this.name} stubs his toe and curses!`;
    if (damage === 1) return `${obj.name} is assailed for ${damage} point of damage.`;
    return `${obj.name} is assailed for ${damage} points of damage.`;
```